src/com/google/enterprise/adaptor/database/UniqueKey.java

```diff
@@ -160,6 +161,57 @@
     return aclSqlCols;
   }
 
+  /**
+   * Attempt to extract ColumnTypes from the supplied Map
+   * of column names to java.sql.Type
+   *
+   * @param sqlTypes Map of column names to Integer java.sql.Types
+   */
+  void addColumnTypes(Map<String, Integer> sqlTypes) {
+    for (Map.Entry<String, Integer> entry : sqlTypes.entrySet()) {
```

Comment:
We can get the Comparator of a TreeMap, and it's null if the natural ordering (aka case-sensitive) is used, so we don't need a separate flag to see if we can apply the SQL types here.